src/db.ts

```diff
@@ -1,23 +1,77 @@
 // db.ts
 import Dexie, { Table } from 'dexie'
 
-export interface Friend {
+export interface Student {
   id?: number
-  name: string
+  first_name: string
+  last_name: string
+  gender: 'male' | 'female' | 'nonbinary' | 'n/a'
   age: number
 }
 
-export class MySubClassedDexie extends Dexie {
+export interface FictionalGrade {
+  id?: number
+  student_id: number
+  v: boolean
+  kd: boolean
+  ca: boolean
+  i: boolean
+  e: boolean
+  l: boolean
+  go: boolean
+  mi: boolean
+}
+
+export interface InformationalGrade {
+  id?: number
+  student_id: number
+  v: boolean
+  kd: boolean
+  ca: boolean
+  i: boolean
+  e: boolean
+  l: boolean
+  go: boolean
+  mi: boolean
+}
+
+export interface SpellingGrade {
+  id?: number
+  student_id: number
+  phonetic_short_vowels: string
+  phonetic_consonant_blends: string
+  phonetic_consonant_digraphs: string
+  transitional_long_vowels: string
+  transitional_complex_vowels: string
+  fluent_inflectional_endings: string
+  fluent_multisyllabic_words_2_syllabes: string
+  advanced_multisyllabic_words_3_syllabes: string
+}
+
+export interface ReadingLevelGrade {
+  student_id: number
+  reading_level: number
+}
+
+export class StudentGradesDB extends Dexie {
   // 'friends' is added by dexie when declaring the stores()
   // We just tell the typing system this is the case
-  friends!: Table<Friend>
+  students!: Table<Student, number>
+  fictional_grades!: Table<FictionalGrade, number>
+  informational_grades!: Table<InformationalGrade, number>
+  spelling_grades!: Table<InformationalGrade, number>
+  reading_grades!: Table<InformationalGrade, number>
 
   constructor() {
-    super('myDatabase')
+    super('StudentGrades')
     this.version(1).stores({
-      friends: '++id, name, age', // Primary key and indexed props
+      students: '++id,[first_name+last_name]',
+      fictional_grades: '++id, student_id',
+      informational_grades: '++id, student_id',
```

Comment:
Referencing the above pr review note, if we want to add a date column, we would also have to reference it here. Just wanted to make that note

Reply:
Answered in [pull/1](https://github.com/phil208-ethics-project/phil208-ethics-project.github.io/pull/2)